### ISSUE TITLE
Weight assignment to objectives

### ISSUE THREAD

**Issue by hdbeukel** [corehunter/corehunter3-ui#6]
It's not clear to me how weights are assigned to the objectives.

1. How are they chosen when adding objectives?
2. How can users change them?

I loaded genotypes and phenotypes and got the two default objectives with with 50%. Adding an additional objective results in a 25-25-50 weighting (why?). Adding one more gives 16-16-33-33. Removing one then yields 20-40-40. What's the logic here?

After following these steps the remove objective button is greyed out while there are still three objectives listed (see #5).

**Comment by guypfr** [comment]
The logic is in correct it seems. I will take a look at it. The idea was to set equal weights to the objective. Users should be able to change them directly in the table. I will test further.

**Comment by daveneti** [comment]
The logic for adding a new objective is contained in these two methods (see below) of the API class in Corehunter. We should agree how these methods are implemented, specifically the second method. At the moment, a new objective is added with weight 1. The second issue is how to view these weights in the interface. At moment it works on viewing a percentage relative to total weight. The bug here is that this is calculated incorrectly. I will fix that. However, displaying a percentage is good, editing a weight and showing a percentage leads to confusion. My suggestion is that we allow the user to change integer weights in one column and show percentage in another column.

```
/**
     * Creates a default allowed objective for the data. If the data contains
     * genotypic data, the default genotypic data objective is returned. If no
     * genotypes are available, then the default phenotypic data objective
     * is returned. Finally, if no genotypic or phenotypic data is available,
     * then the default distances data objective is returned.
     * 
     * @param coreHunterData the data for which the objective is required
     * @return a default objective (with weight 1.0) or null if the data is not defined
     */
    public static final CoreHunterObjective createDefaultObjective(CoreHunterData coreHunterData) {

        if (coreHunterData != null) {

            if (coreHunterData.hasGenotypes()) {
                return new CoreHunterObjective(DEFAULT_OBJECTIVE, DEFAULT_GENOTYPE_MEASURE);
            }
            if (coreHunterData.hasPhenotypes()) {
                return new CoreHunterObjective(DEFAULT_OBJECTIVE, CoreHunterMeasure.GOWERS);
            }
            if (coreHunterData.hasDistances()) {
                return new CoreHunterObjective(DEFAULT_OBJECTIVE, CoreHunterMeasure.PRECOMPUTED_DISTANCE);
            }
        }

        return null;
    }
    
    /**
     * Creates a default allowed objective for the data given a list existing
     * objectives.
     * 
     * @param coreHunterData
     *            the data for which the objective is required
     * @param currentObjectives
     *            a list existing objectives
     * @return a default objective (with weight 1.0), or null if all possible
     *         objectives used or the data is not defined
     */
    public static final CoreHunterObjective createDefaultObjective(CoreHunterData coreHunterData, 
        List<CoreHunterObjective> currentObjectives) {

        if (coreHunterData != null) {
             
            Iterator<CoreHunterObjective> possibleObjectives = getAllAllowedObjectives(coreHunterData).iterator() ;
            
            boolean found = false ;
            
            CoreHunterObjective objective = null ;
            CoreHunterObjective currentObjective ;
            Iterator<CoreHunterObjective> iterator ;
            
            while (!found && possibleObjectives.hasNext()) {
                objective = possibleObjectives.next() ;
                
                iterator = currentObjectives.iterator() ;
                
                boolean alreadyUsed = false ;
                
                while (!alreadyUsed && iterator.hasNext()) {
                    currentObjective = iterator.next() ;
                    
                    alreadyUsed = objective.isSameObjective(currentObjective) ;
                }
                
                found = !alreadyUsed ;
            }

            return objective ;
        }

        return null;
    }
```

**Comment by daveneti** [comment]
Ok, it seems the method   `API.createDefaultObjectives(CoreHunterData coreHunterData)` is inconsistent with the above methods. When the objectives are created for a data object in first use the weights are normalised to 0-1, whereas if an objective is added later it is not normalised. I propose to change `API.createDefaultObjectives(CoreHunterData coreHunterData)` so that all objectives are set to 1 default. Interface can then use only integers for weights to simply everything. With the changes made to the UI, this issue is closed in favour of corehunter/corehunter#86
